// react-simple-form-validator/lib/cjs/rules/defaultRules.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var dayjs_1 = __importDefault(require("dayjs"));
// Custom default rules to validate form fields
var defaultRules = {
    numbers: /^(([0-9]*)|(([0-9]*)\.([0-9]*)))$/,
    email: /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
    required: /\S+/,
    hasNumber: /\d/,
    hasUpperCase: /(?=.*[A-Z])/,
    hasLowerCase: /(?=.*[a-z])/,
    hasSpecialCharacter: /(\W)/,
    date: function (format, value) {
        if (format === void 0) { format = 'YYYY-MM-DD'; }
        var d = (0, dayjs_1.default)(value, format);
        if (d == null || !d.isValid())
            return false;
        return true;
    },
    minlength: function (length, value) {
        if (length === void 0) {
            throw 'ERROR: It is not a valid length, checkout your minlength settings.';
        }
        else if (value.length >= length) {
            return true;
        }
        return false;
    },
    maxlength: function (length, value) {
        if (length === void 0) {
            throw 'ERROR: It is not a valid length, checkout your maxlength settings.';
        }
        else if (value.length > length) {
            return false;
        }
        return true;
    },
    equalPassword: function (dataToCompare, value) {
        return dataToCompare === value;
    }
};
exports.default = defaultRules;
